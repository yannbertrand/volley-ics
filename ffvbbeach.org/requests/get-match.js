import PDFParser from 'pdf2json'
import getRequestBody from '../body.builder.js'

export async function getMatch(client, saison, ent, matchId) {
  let bufferResult
  try {
    bufferResult = await new Promise(async (resolve, reject) => {
      let buff = new Buffer.alloc(0)
      const { body } = await client.request({
        path: '/ffvbapp/adressier/fiche_match_ffvb.php',
        method: 'POST',
        headers: {
          'content-type': 'application/x-www-form-urlencoded',
        },
        body: getRequestBody({
          wss_saison: saison,
          codent: ent,
          codmatch: matchId,
        }),
      })
      body.on('error', (error) => {
        reject(error)
      })
      body.on('data', (chunk) => {
        buff = Buffer.concat([buff, chunk])
      })
      body.on('end', () => {
        resolve(buff)
      })
    })
  } catch (exception) {
    // ToDo
    console.error(exception)
    throw new Error(
      'An unkown error occured while requesting www.ffvbbeach.org'
    )
  }

  const pages = await new Promise((resolve, reject) => {
    const pdfParser = new PDFParser()
    const tmpPages = []
    pdfParser.on('readable', (meta) => console.log('PDF Metadata', meta))
    pdfParser.on('data', (page) => {
      if (page === null) {
        return resolve(tmpPages)
      }

      tmpPages.push(
        page.Texts.flatMap((t) => t.R.map((R) => decodeURIComponent(R.T)))
      )
      console.log(page)
    })
    pdfParser.on('error', (error) => reject(error))

    pdfParser.parseBuffer(bufferResult)
  })

  return pages[0]
}
