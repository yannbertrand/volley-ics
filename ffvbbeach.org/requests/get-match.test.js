import { beforeEach, describe, expect, it } from 'vitest'
import getMockedFfvbbClient from '../mock.js'
import { getMatch } from './get-match.js'

describe.only('getSaisonMatches', () => {
  let matchAsptt1vsAsbr1
  beforeEach(async () => {
    client = await getMockedFfvbbClient()
    matchAsptt1vsAsbr1 = await getMatch(
      client,
      '2021/2022',
      'PTPL44',
      'LA1A021'
    )
  })

  it('should fetch ffvbbeach.org data', () => {
    expect(matchAsptt1vsAsbr1).toBeDefined()
    expect(matchAsptt1vsAsbr1).toMatchSnapshot()
  })
})
