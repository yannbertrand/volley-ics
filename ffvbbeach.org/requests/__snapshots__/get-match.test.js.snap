// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getMatch > should fetch ffvbbeach.org data 1`] = `
[
  " 2021/2022",
  "POULE A1 COMPETLIB - LA1",
  "ASPTT 1",
  "-----",
  "ASBR 1",
  "Match: LA1A021 - Lundi 28 Février 2022 à 20h30",
  "Salle",
  "SALLE LONGCHAMP",
  "RUE PINEAU CHAILLOU  ",
  "44300 NANTES",
  "Tél.: 0240494607",
  "Arbitre(s) ",
  "A1. ",
  "A2. ",
  "ASPTT NANTES",
  "3 - 0442446",
  "Ligue PAYS DE LA LOIRE",
  "   |   M. PAPIN MANUEL (Pdt)",
  "Correspondant :",
  "M. PAPIN MANUEL",
  "4 RUE CHARLES TERRONT",
  "44300 NANTES",
  "Port.: ",
  "06.76.39.01.11",
  "Mél.: ",
  "manuel.papin@laposte.net",
  "Salle principale :",
  "SALLE LONGCHAMP",
  "RUE PINEAU CHAILLOU",
  "44300 NANTES",
  "Tél.: ",
  "0240494607",
  "Salle de repli :",
  " ",
  "ASBR VOLLEY-BALL DETENTE",
  "10 - 0442443",
  "Ligue PAYS DE LA LOIRE",
  "   |   M. PILORGE JACQUES (Pdt)",
  "Correspondant :",
  "PILORGE JACQUES",
  "8 RUE DE RUTHERFORD",
  "44400 REZE",
  "Port.: ",
  "06.63.53.83.95",
  "Mél.: ",
  "asbrvolleyballdetente@gmail.com",
  "Salle principale :",
  "GYMNASE L. ET M. LIBERGE",
  "11 RUE DE LA GALARNIERE",
  "44400 REZE",
  "Tél.: ",
  "0240754858",
  "Salle de repli :",
  " ",
  "Loire-Atlantique",
  "44 RUE ROMAIN ROLLAND   ",
  "44100 NANTES",
  "volley44@orange.fr",
]
`;
